cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/configs/StorageProperties.java

```diff
@@ -0,0 +1,35 @@
+package com.linkedin.openhouse.cluster.storage.configs;
+
+import com.linkedin.openhouse.cluster.configs.YamlPropertySourceFactory;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.*;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+
+@Configuration
+@ConfigurationProperties(prefix = "cluster.storages")
+@PropertySource(
+    name = "clusterStorage",
+    value = "file:${OPENHOUSE_CLUSTER_CONFIG_PATH:/var/config/cluster.yaml}",
+    factory = YamlPropertySourceFactory.class,
+    ignoreResourceNotFound = true)
+@Getter
+@Setter
+public class StorageProperties {
+  private String defaultType;
+  private Map<String, StorageTypeProperties> types = new HashMap<>();
```

Comment:
it is actually not common to instantiate field variable outside of ctor especially if it is not a static. curious why did we decide to do so, instead of using build or ctor?

Reply:
following [previous code practice](https://github.com/linkedin/openhouse/blob/main/services/jobs/src/main/java/com/linkedin/openhouse/jobs/config/JobsProperties.java), its not necessary, will address.